src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2316,22 +2316,72 @@ private static void AddNestedTypesToDictionary(Dictionary<string, ImmutableArray
         private class MembersAndInitializersBuilder
         {
             public ArrayBuilder<Symbol> NonTypeNonIndexerMembers { get; private set; } = ArrayBuilder<Symbol>.GetInstance();
-            public readonly ArrayBuilder<ImmutableArray<FieldOrPropertyInitializer>> StaticInitializers = ArrayBuilder<ImmutableArray<FieldOrPropertyInitializer>>.GetInstance();
-            public readonly ArrayBuilder<ImmutableArray<FieldOrPropertyInitializer>> InstanceInitializers = ArrayBuilder<ImmutableArray<FieldOrPropertyInitializer>>.GetInstance();
+            public readonly ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>> StaticInitializers = ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>>.GetInstance();
+            public readonly ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>> InstanceInitializers = ArrayBuilder<ArrayBuilder<FieldOrPropertyInitializer.Builder>>.GetInstance();
             public readonly ArrayBuilder<SyntaxReference> IndexerDeclarations = ArrayBuilder<SyntaxReference>.GetInstance();
+            public ParameterListSyntax? RecordConstructorParamList;
+            public TypeDeclarationSyntax? RecordDeclarationWithParameters;
+            public ArrayBuilder<FieldOrPropertyInitializer.Builder>? InstanceInitializersForRecordDeclarationWithParameters;
```

Comment:
>InstanceInitializersForRecordDeclarationWithParameters [](start = 69, length = 54)

It seems `InstanceInitializersForRecordDeclarationWithParameters` is only ever read in debug-only code. Can we remove it or make it debug-only?

Reply:
> It seems InstanceInitializersForRecordDeclarationWithParameters is only ever read in debug-only code. Can we remove it or make it debug-only? 

Of course it is read, otherwise record properties and other fields with initializers wouldn't be initialized. This instance is in the InstanceInitializers builder.

---
In reply to: [434398822](https://github.com/dotnet/roslyn/pull/44803#discussion_r434398822) [](ancestors = 434398822)